protocol/validation/tx.go

```diff
@@ -48,6 +48,13 @@ func ConfirmTx(snapshot *state.Snapshot, block *bc.Block, tx *bc.Tx) error {
 			if txin.AssetVersion != 1 {
 				continue
 			}
+			if ii.InitialBlock != initialBlockHash {
+				return errors.WithDetail(ErrBadTx, "issuance is for different blockchain")
+			}
+			assetID := bc.ComputeAssetID(ii.IssuanceProgram, initialBlockHash, ii.VMVersion)
```

Comment:
super tiny nit: should we use `ii.InitialBlock` instead of `initialBlockHash`? When we support joining chains, I think this would need to use `ii.InitiialBlock` and the above check would check that `ii.InitialBlock` is in a set of acceptable initial block hashes